curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/63fc93935f0331b591cc65e8.md

```diff
@@ -0,0 +1,138 @@
+---
+id: 63fc93935f0331b591cc65e8
+title: Step 2
+challengeType: 0
+dashedName: step-2
+---
+
+# --description--
+
+You will now get into using the `fetch` API. It is a built-in JavaScript interface for getting a resource from a server. It has a `fetch()` method with which you can make a `GET`, `POST`, `PUT`, or `PATCH` request. The request you are going to make in this project is a `GET` request to this endpoint `https://cdn.freecodecamp.org/curriculum/news-author-page/authors.json`. The endpoint has the information of freeCodeCamp News Site authors – making it ideal for what you want to build.
+
+To make a `GET` request with the `fetch()` method, you do it like this:
+
+```js
+fetch("sample-endpoint-goes-here")
+```
+
+The `fetch()` method returns a `Promise`, which is an important aspect of asynchronous JavaScript. The `Promise` is either `fulfilled` or `rejected`. You can use the `.then()` method to access the `Promise` if it fulfils and the `.catch()` method to handle errors when it rejects.
```

Comment:
It might be best to introduce this concept in the next step, and go a bit more into what a `Promise` is (a placeholder object that either resolves to a `Response` object, or is rejected, etc.).

Reply:
Thanks for pointing that out, Kris. I think it makes more sense to introduce it in the next step. But that's the part you talked about having a long description.

I shortened the text a little. And now adding this would make it long again.

Reply:
This is what it looks like now:

```
The `fetch()` method returns a `Promise`, which is  placeholder object that either resolves to a `Response` object, or is rejected. You can use the `.then()` method to access the `Promise` if it fulfils, and the `.catch()` method to handle errors when it rejects.

And to make the data you get from a `GET` request usable, you can use the `json()` method on the response with the `.then()` method.

You can chain a `.then()` to the `fetch()` method this way:

```js
fetch("sample-url-goes-here")
  .then((res) => res.json())
```

Chain the `.then()` method to your fetch call. Inside the `.then()` method, add a callback function with a `res` parameter that returns `res.json()`. You don't need to use curly braces because the arrow function will have one expression.

Again, don't terminate the code with a semi-colon yet. 
```

Is that okay?

Reply:
> Hey @Ksound22, this is looking great so far.
> 
> I didn't get a chance to go through every step in detail, but I left some general comments for the first handful of steps and the concepts you cover in those.

Thanks, Kris. I'm attending to them now.

Reply:
I figured out we can also log the `authorDataArr` array to the console – that's after we assign the `data` to it. Would it make sense to make it look like this:

```js
console.log('Author Data Array:', authorDataArr);
```

Just to let campers know its the same as the data – some kind of confirmation.